pkg/resource/model_package_group/hooks.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package model_package_group
+
+import (
+	"context"
+	"errors"
+
+	ackcondition "github.com/aws-controllers-k8s/runtime/pkg/condition"
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	svcsdk "github.com/aws/aws-sdk-go/service/sagemaker"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	requeueWaitWhileDeleting = ackrequeue.NeededAfter(
+		errors.New("ModelPackageGroup is deleting"),
+		ackrequeue.DefaultRequeueAfterDuration,
+	)
+)
+
+func (rm *resourceManager) customSetOutput(r *resource) {
+	if r.ko.Status.ModelPackageGroupStatus == nil {
+		return
+	}
+	ModelPackageGroupStatus := *r.ko.Status.ModelPackageGroupStatus
+	msg := "ModelPackageGroup is in" + ModelPackageGroupStatus + "status"
+	if !isModifiying(r) {
+		ackcondition.SetSynced(r, corev1.ConditionTrue, &msg, nil)
+	} else {
```

Comment:
nit: drop the else 

```
if !isModifiying(r) {
  ackcondition.SetSynced(r, corev1.ConditionTrue, &msg, nil)
}
ackcondition.SetSynced(r, corev1.ConditionFalse, &msg, nil)
```

Reply:
@surajkota  need the else statement since if it is NOT modifying it will set Synced condition to True otherwise it will set to false. Without the else statement if it is not modifying Synced will be set to True but then will immediately be set to false.